src/native/libs/System.Native/pal_datetime.c

```diff
@@ -21,7 +21,7 @@ static const int64_t TICKS_PER_MICROSECOND = 10; /* 1000 / 100 */
 #endif
 
 #if defined(TARGET_WASI) || defined(TARGET_BROWSER)
-extern const unsigned char* mono_wasm_get_bundled_file(const char* name, int* out_length);
+extern MonoBundledResource * mono_get_bundled_resource_data (const char *id);
```

Comment:
External components, like System.Native won't have access to the layout of MonoBundledResource and MonoBundledDataResource and should not include internal runtime specific header files. Maybe we should add an additional API where data and len can be queried without the need to get hold of the struct first, like:

bool mono_get_bundled_resource_data_values (const char *id, const uint8_t **data_out, uint32_t *size_out)

That would simplify the external usage of the API, to just be an extern declare of standard C99 types.